core/serialization/deserializer/montage-reviver.js

```diff
@@ -687,14 +689,9 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
 
     didReviveObjects: {
         value: function (objects, context) {
-            var self = this;
-
-            return Promise.all([
-                this._deserializeBindings(context),
-                this._deserializeUnits(context)
-            ]).then(function () {
-                self._invokeDeserializedFromSerialization(objects, context);
-            });
+            this._deserializeBindings(context);
```

Comment:
Where are the promises in async case?

Reply:
In practice, both _deserializeBindings and _deserializeUnits don't do any async work. So instead of creating extra promises for no reason, we can make this method synchronous.

Reply:
In practice, both _deserializeBindings and _deserializeUnits don't do any async work. So instead of creating extra promises for no reason, we can make this method synchronous.